api/MetricsApi.ts

```diff
@@ -14,8 +14,20 @@ async function findAll(): Promise<MetricBare[]> {
   return (await fetchApi('GET', '/metrics')).metrics.map((apiData: ApiData) => MetricBare.fromApiData(apiData))
 }
 
+/**
+ * Find the metric by ID.
+ *
+ * Note: Be sure to handle any errors that may be thrown.
+ *
+ * @throws UnauthorizedError
+ */
+async function findById(metricId: number): Promise<MetricFull> {
+  return MetricFull.fromApiData(await fetchApi('GET', `/metrics/${metricId}`))
+}
+
 const MetricsApi = {
   findAll,
+  findById,
 }
 
 export default MetricsApi
```

Comment:
Any reason to not use:

```suggestion
export default {
  findAll,
  findById,
}
```

Reply:
The namespacing helps with self-documentation and avoids name conflicts. If I come across just `findAll`, then I need to see which module it is coming from. I just tried it in VSCode and tested what information would be available on hover. If the comment and return types are adequate, then you can figure out the which function it is. But this does not work during a code review.

Also, we are using `findAll` and `findById` in other `*Api.ts` modules that would conflict. There are ways around this conflict with aliases, but I think the namespacing provides the best DX overall.

Reply:
+1 to namespacing, it make things clearer. :slightly_smiling_face: